### PR TITLE
SYS-3338 Finalises NFT manager data migration to bounded data structures

### PR DESCRIPTION
**Pull request by thadouk**
## Proposed changes
[Diff with 3.1.1 release](https://github.com/Aventus-Network-Services/avn-parachain/compare/v3.1.1...656507f1e686fd27d736d7aa20b08dcb441756f0)
Reintroduces commit b4dfc4249bee3f50f5d73c9b5b7ea1910224e684 which finalises the NFT manager data migration.
Revert the stopgap solution for the migration and the migration code itself.

## Type of change/Merge

🚨What type of change is this PR? </br>
_Put an `x` in the boxes that apply_

- [x] Release 
  - [x] Increase versions 
  - [ ] Baseline tests passed  
  - Release type:
    - [ ] Major release 
    - [ ] Minor release 
    - [x] Patch release 

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR._

- [ ] You describe the purpose of the PR, e.g.:
  - What does it do?
  - Highlight what important points reviewers should know about;
  - Indicates if there is something left for follow-up PRs.
- [ ] Documentation updated
- [ ] Business logic tested successfully
- [ ] Verify First, Write Last: In Substrate development, it is important that you always ensure preconditions are met and return errors at the beginning. After these checks have completed, then you may begin the function's computation.

### CODE REVIEW

**Comment by alexpArtos** [comment]
I don't see any changes in `cargo.toml` but the diff with 3.1.1 throws this up: shouldn't we be upgrading the version here to `3.2.1` instead of `3.2.0`?

**Review by alexpArtos — APPROVED** [review]
Approved, but with comments on the main thread (not the code)

**Review by ivan-cholakov — APPROVED** [review]

**Comment by thadouk** [comment]
> I don't see any changes in `cargo.toml` but the diff with 3.1.1 throws this up: shouldn't we be upgrading the version here to `3.2.1` instead of `3.2.0`?
> 
> And does the spec_version need to stay the same as well?

You are right. It was updated when the release was enabled.

File: pallets/nft-manager/src/lib.rs [hunk 1]
```diff
@@ -774,37 +771,6 @@ pub mod pallet {
 
             Ok(())
         }
-
-        #[pallet::weight(T::DbWeight::get().reads_writes(1 as u64, 2 as u64))]
```
Comment (thadouk) [inline pallets/nft-manager/src/lib.rs:778]: TODO:
Ensure this is completely safe and that the call_index can be reused. If not, find out how to "burn it"

**Review by MBrozhko34 — APPROVED** [review]

**Review by alexpArtos — APPROVED** [review]

File: pallets/nft-manager/src/lib.rs (hunk 1 above)
Comment (thadouk) [reply pallets/nft-manager/src/lib.rs:778]: Based on [this discussion on stackexchange](https://substrate.stackexchange.com/questions/8923/safe-way-to-remove-an-extrinsic):
We can remove the extrinsic and reuse the index provided we give enough time to clients to upgrade. It is safe though in the blockchain context.

**Pull request merged**